### PR TITLE
tracking bio relies on submit_bio instead ftrace

### PR DESCRIPTION
**Pull request by Swistusmen**
replacement the attitude which used ftrace to track bio. Now we are relying on block_device_operations and it's submit bio

### CODE REVIEW

File: src/bio_helper.h [hunk 1]
```diff
@@ -28,6 +28,12 @@
         // block_device_operations as submit_bio function.
         // See https://github.com/torvalds/linux/commit/c62b37d96b6eb3ec5ae4cbe00db107bf15aebc93
         #define USE_BDOPS_SUBMIT_BIO
+
```
Comment (natalia-zelazna) [inline src/bio_helper.h:31]: mrf_type.h proposed on branch https://github.com/natalia-zelazna/dattobd/tree/addr_brutforce_test allows to cut off from this ifdef, we can consider it once this PR is closed

File: src/configure-tests/symbol-tests [hunk 2]
```diff
@@ -3,3 +3,5 @@ sys_umount
 sys_oldumount
 sys_call_table
 printk
+blk_mq_submit_bio
+kfree
```
Comment (natalia-zelazna) [inline src/configure-tests/symbol-tests:7]: printk became a macro, so it can be removed from this file asit always has addr zero, other have to be checked

Comment (Swistusmen) [reply src/configure-tests/symbol-tests:6]: resolved

File: src/bio_helper.h (hunk 1 above)
Comment (Swistusmen) [reply src/bio_helper.h:31]: I think we should make a refactoring ticket to try to get rid of ifdefs, not only this on

**Review by natalia-zelazna — APPROVED** [review]

File: src/tracer.c [hunk 3]
```diff
@@ -1168,57 +1190,6 @@ static int __tracer_setup_cow_thread(struct snap_device *dev,
         return ret;
 }
 
-#ifdef USE_BDOPS_SUBMIT_BIO
-#ifdef HAVE_FTRACE_REGS
```
Comment (natalia-zelazna) [inline src/tracer.c:1172]: I really like the amount of code removed :D

**Pull request merged**